numba_dpex/core/caching.py

```diff
@@ -0,0 +1,387 @@
+# Copyright 2022 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import hashlib
+from abc import ABCMeta, abstractmethod
+
+from numba.core.caching import CacheImpl, IndexDataCacheFile, _Cache
+from numba.core.serialize import dumps
+
+from numba_dpex import config
+
+
+def build_key(sig, pyfunc, codegen, backend=None, device_type=None):
+    """Constructs a key from the data object.
+
+    Compute index key for the given signature and codegen. It includes
```

Comment:
Does not seem right. Where are we passing OS information?

Reply:
We are building a key with backend and device_type. This function is called in dispatcher (before caching/decaching). But where do I get OS info?

Reply:
We do not need this.